### PR TITLE
Reference docs generation and plan.

### PR DESCRIPTION
**Pull request by mixmix**
mega PR for us to jam on till it's done _(as discussed with Frankie)_

The aim of this PR is to :
- makes `README.md` a chill into + link out to further resources 
- build `docs/` into a rich website

What the website will look like:

image

### CODE REVIEW

**Comment by frankiebee** [comment]
This was a notes bucket... @mixmix anything worth keeping cc @frankiebee

**Comment by mixmix** [comment]
Here's what it looks like now with current config
image

**Comment by mixmix** [comment]
@frankiebee / @rh0delta this is ready for a review.
I through out a lot of stuff and have just kept the  "docs building" piece which makes HTML site which @johnnymatthews will presumably integrate at some place??

One thing we need to decide is what we want that site to look like. Currently that ugly HTML page has an ... on site map nested thing on the left but it's ... awkwardly named. Like it doesn't line up with what we export from the `package.json` :vomiting_face: So ... do we care/ wanna fix it? I've looked into typedoc config and can't find a nicer way of renaming those other than "move all the files" ... I think we might need to discuss "design" around what we export and why?

**Comment by mixmix** [comment]
Decision: we need to have a documentation meeting
@johnnymatthews can you come up with a  proposal about how we're doing docs this / next week?

**Comment by mixmix** [comment]
@frankiebee please review

**Comment by mixmix** [comment]
@johnnymatthews can we merge this?

**Comment by mixmix** [comment]
Found a plugin which makes the entry points (sections on left) less of a mess

image

**Comment by mixmix** [comment]
Hey I want this PR merged or closed this week.
@johnnymatthews I propose we merge as is, and set up Github generation of static site. We can change it later if we want

File: dev/README.md [hunk 1]
```diff
@@ -1,5 +1,31 @@
...
+This is your home base for how to develop on the project locally and process around that
+
+## Branches
+
+- `dev`
+  in development
+- `main`
+  "stable" what is on npm
+
+- `<name>/<issue_number>-<short_title>` (personal branches)
```
Comment (rh0delta) [inline dev/README.md:12]: Is issue number required for the branch name? Will this now be the new standard? More often than not, our branches don't include the issue number in the name, but rather required in the PR description prefixed by "closes" or "fixes" or something along those lines

File: dev/README.md [hunk 2]
```diff
@@ -1,5 +1,31 @@
...
+  for example `frankie/i123-docs`
+
+all PRS go into `dev`
```
Comment (rh0delta) [inline dev/README.md:15]: Should we make a comment about hotfix prs going straight to main?

File: dev/README.md [hunk 3]
```diff
@@ -1,5 +1,31 @@
...
+
+## Versions
+
+The SDK version must match a compatible [`entropy-core`](https://github.com/entropyxyz/entropy-core) version.
+
+<!-- TODO: -->
```
Comment (johnnymatthews) [inline dev/README.md:21]: What's this here for?

Comment (mixmix) [reply dev/README.md:21]: An invisible comment!

i think I'd like to see a thing there where we list the details of the compatible versions of things. 

this was just from using other sdks where it became a nightmare to know how to run a compatible network of tools cos the versions of things which work together was not well documented.
i may have trauma

File: dev/README.md (hunk 1 above)
Comment (mixmix) [reply dev/README.md:12]: I think I copied this from notes from a call with frankie. I'm not sure I care about issue number.
Let's discuss

File: dev/README.md (hunk 2 above)
Comment (mixmix) [reply dev/README.md:15]: nice. Those would be patch releases aye?

i feel like we could make a patch branch. Basically anything we write ha gotta get smoke testing befor going into main + published...

depends how many patches we expect. What I'm being cautious of is "special shorcuts". They can cause problems

**Review by frankiebee — APPROVED** [review]

**Review by rh0delta — APPROVED** [review]

**Pull request merged**